### PR TITLE
Added support for listing cups printers

### PR DESCRIPTION
**Pull request by BlackHeart-TF**
Added GetCUPSPrinters request to list available printers on an CUPS server

### CODE REVIEW

File: SharpIpp/Model/CUPSGetPrintersRequest.cs [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+        public int? PrinterID { get; set; }
+
+        ///The client OPTIONALLY supplies this attribute to select which printers are returned.
+        public string? PrinterLocation { get; set; }
+
+
+        ///The client OPTIONALLY supplies a printer type enumeration to select which printers are returned.
+        public PrinterType? PrinterType { get; set; }
+
+        ///The client OPTIONALLY supplies a printer type mask enumeration to select which bits are used in the "printer-type" attribute.
```
Comment (Zelenov) [inline SharpIpp/Model/CUPSGetPrintersRequest.cs:48]: Could you please put this remarks either into `<summary>` or into `<remarks>` block?

File: SharpIpp/Model/CUPSGetPrintersRequest.cs [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+        ///     The "requesting-user-name" (name(MAX)) attribute SHOULD be
+        ///     supplied by the client
+        /// </summary>
+        public string? RequestingUserName { get; set; }
+
+        public IEnumerable<IppAttribute>? AdditionalOperationAttributes { get; set; }
+        public IEnumerable<IppAttribute>? AdditionalJobAttributes { get; set; }
+    }
+
+    public enum PrinterType
```
Comment (Zelenov) [inline SharpIpp/Model/CUPSGetPrintersRequest.cs:86]: `PrinterType = Printer`? What does it mean? Please, rename it or add `summary` comment

File: SharpIpp/Protocol/IppProtocol.GetCUPSPrinters.cs [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+        {
+            mapper.CreateMap<CUPSGetPrintersRequest, IppRequestMessage>((src, map) =>
+            {
+                var dst = new IppRequestMessage {IppOperation = IppOperation.GetCUPSPrinters};
+                mapper.Map<IIppPrinterRequest, IppRequestMessage>(src, dst);
+                var operation = dst.OperationAttributes;
+                if (src.Limit != null)
+                    operation.Add(new IppAttribute(Tag.Integer, "requesting-user-name", src.Limit.Value));
+                if (src.FirstPrinterName != null)
+                    operation.Add(new IppAttribute(Tag.Keyword, "first-printer-name", Mapper.Map<string>(src.FirstPrinterName)));
```
Comment (Zelenov) [inline SharpIpp/Protocol/IppProtocol.GetCUPSPrinters.cs]: Use local `mapper` variable

File: SharpIpp/Model/CUPSGetPrintersRequest.cs (hunk 2 above)
Comment (BlackHeart-TF) [reply SharpIpp/Model/CUPSGetPrintersRequest.cs:86]: The docs just say its an enum with no hint to possible values. "Printer" is all i've seen come back in practise

Comment (Zelenov) [reply SharpIpp/Model/CUPSGetPrintersRequest.cs:86]: This are the docs for this flags: https://www.cups.org/doc/spec-ipp.html#printer-type
Seems like you have to make this thing a `[Flags]` enum. And add mapping for it
```csharp
static IppProtocol()
{
    ...
    Mapper.CreateIppMap<int, PrinterType>((src, map) => (PrinterType) src);
}
```

Comment (BlackHeart-TF) [reply SharpIpp/Model/CUPSGetPrintersRequest.cs:86]: I spent an hour yesterday trying to find that and i feel like you probably found it in 2 minutes.
It should be good now!

**Pull request merged**